server/src/services/initialization/onInitialized.ts

```diff
@@ -33,45 +29,16 @@ export const onInitialized = (
       });
     }
 
+    // index folders
     await serverState.telemetry.trackTiming("indexing", async () => {
       await indexWorkspaceFolders(
-        { ...serverState, workspaceFolders: [] },
+        serverState,
         workspaceFileRetriever,
-        serverState.workspaceFolders
+        serverState.workspaceFoldersToIndex
       );
+      serverState.indexingFinished = true;
```

Comment:
Is this check needed? It is used as a guard against premature validation and other actions right? Should we move the indexing to the initialization, and only return from initialization when all the projects have initialized?

Reply:
Lets add a detailed linear task for this.